Lib/Emscripten/Emscripten.cpp

```diff
@@ -163,6 +164,10 @@ DEFINE_INTRINSIC_FUNCTION(env, "_time", I32, _time, U32 address)
 
 DEFINE_INTRINSIC_FUNCTION(env, "___errno_location", I32, ___errno_location) { return 0; }
 
+DEFINE_INTRINSIC_FUNCTION(env, "___setErrNo", void, ___seterrno, I32 value) {
+	if(emscriptenErrNoLocation) { memoryRef<I32>(emscriptenMemory, emscriptenErrNoLocation) = (I32)value; }
```

Comment:
Full disclosure: I didn't really test this part (write to `___errno_location`), but it looks fine.